backend/nokia-innovative-project/src/main/java/com/nokia/library/nokiainnovativeproject/services/BookDetailsService.java

```diff
@@ -2,59 +2,93 @@
 
 import com.nokia.library.nokiainnovativeproject.DTOs.BookDetailsDTO;
 import com.nokia.library.nokiainnovativeproject.entities.Author;
+import com.nokia.library.nokiainnovativeproject.entities.BookCategory;
 import com.nokia.library.nokiainnovativeproject.entities.BookDetails;
 import com.nokia.library.nokiainnovativeproject.exceptions.ResourceNotFoundException;
+import com.nokia.library.nokiainnovativeproject.repositories.AuthorRepository;
+import com.nokia.library.nokiainnovativeproject.repositories.BookCategoryRepository;
 import com.nokia.library.nokiainnovativeproject.repositories.BookDetailsRepository;
 import lombok.RequiredArgsConstructor;
+import org.hibernate.Hibernate;
 import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.logging.Logger;
 
 @Service
 @RequiredArgsConstructor
 public class BookDetailsService {
 
 	private final BookDetailsRepository bookDetailsRepository;
+	private final AuthorRepository authorRepository;
+	private final BookCategoryRepository bookCategoryRepository;
 
 	public List<BookDetails> getAllBookDetails() {
-		return bookDetailsRepository.findAll();
+		List<BookDetails> list = bookDetailsRepository.findAll();
+		for(BookDetails bookDetails : list) {
+			Hibernate.initialize(bookDetails.getAuthors());
+			Hibernate.initialize(bookDetails.getCategories());
+			Hibernate.initialize(bookDetails.getReviews());
+			Hibernate.initialize(bookDetails.getBooks());
+		}
+		return list;
 	}
 
 	public BookDetails getBookDetailsById(Long id) {
-		return bookDetailsRepository.findById(id).orElseThrow(() -> new ResourceNotFoundException("BookDetailsRepository", "id", id));
+		BookDetails bookDetails = bookDetailsRepository.findById(id).orElseThrow(() -> new ResourceNotFoundException("book details"));
+		Hibernate.initialize(bookDetails.getAuthors());
+		Hibernate.initialize(bookDetails.getCategories());
+		Hibernate.initialize(bookDetails.getReviews());
+		Hibernate.initialize(bookDetails.getBooks());
+		return bookDetails;
 	}
 
 	public BookDetails createBookDetails(BookDetailsDTO bookDetailsDTO) {
 		ModelMapper mapper = new ModelMapper();
 		BookDetails bookDetails = mapper.map(bookDetailsDTO, BookDetails.class);
-		bookDetails.setAuthors(new ArrayList<>());
-		bookDetails.setCategories(new ArrayList<>());
 
-		BookDetails objectDetails = bookDetailsRepository.save(bookDetails);
+		List<Author> authors = bookDetailsDTO.getAuthors();
+		if(authors != null && authors.size() > 0) {
+			for(Author author : authors) {
+				if(author.getId() != null){
+					authors.remove(author);
+					authors.add(authorRepository.findById(author.getId()).orElseThrow(()-> new ResourceNotFoundException("author")));
+				}
+			}
+		}
```

Comment:
Authors is @NotNull, no reason to check that second time. Similarly with Size. if size is 0 then the loop won't run anyway